pkg/util/ringlog/ringlog.go

```diff
@@ -0,0 +1,84 @@
+package ringlog
+
+import (
+	"container/list"
+	"sync"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+
+	"github.com/zalando-incubator/postgres-operator/pkg/spec"
+)
+
+// RingLogger describes ring logger methods
+type RingLogger interface {
+	Insert(level logrus.Level, t time.Time, workerID *uint32, clusterName spec.NamespacedName, message string) error
+	Walk() []*LogEntry
+}
+
+// RingLog is a capped logger with fixed size
+type RingLog struct {
+	size int
+	list *list.List
+	mu   *sync.RWMutex
```

Comment:
Does it make sense to embed *sync.RWMutex to be able to write r.Lock instead of r.mu.Lock?